helm/cluster-azure/templates/_machine_deployments.tpl

```diff
@@ -0,0 +1,71 @@
+{{- define "machine-deployments" -}}
+{{- range $machineDeployment := .Values.machineDeployments }}
+{{ $data := dict "spec" $machineDeployment "type" "machineDeployment" "Values" $.Values "Release" $.Release "Files" $.Files "Template" $.Template }}
+{{ $kubeAdmConfigTemplateHash := dict "hash" ( include "hash" (dict "data" (include "machine-kubeadmconfig-spec" $data) "global" $) ) }}
+{{ $azureMachineTemplateHash := dict "hash" ( include "hash" (dict "data" ( dict "spec" (include "machine-spec" $data) "identity" (include "machine-identity" $data) ) "global" $) ) }}
+apiVersion: cluster.x-k8s.io/v1beta1
+kind: MachineDeployment
+metadata:
+  annotations:
+    machine-deployment.giantswarm.io/name: {{ include "resource.default.name" $ }}-{{ .name }}
+  labels:
+    giantswarm.io/machine-deployment: {{ include "resource.default.name" $ }}-{{ .name }}
+    {{- include "labels.common" $ | nindent 4 }}
+  name: {{ include "resource.default.name" $ }}-{{ .name }}
+  namespace: {{ $.Release.Namespace }}
+spec:
+  clusterName: {{ include "resource.default.name" $ }}
+  replicas: {{ .replicas }}
+  selector:
+    matchLabels: null
```

Comment:
i assume we can drop these two lines
```suggestion
```

Reply:
yeah probably you are right, this is in every example online but i will try this morning without this

Reply:
looks good

```
➜ k get machinedeployment fctest1-md00 -o json | jq .spec.selector
{
  "matchLabels": {
    "cluster.x-k8s.io/cluster-name": "fctest1",
    "cluster.x-k8s.io/deployment-name": "fctest1-md00"
  }
}
```

Reply:
actually i did not commit this change :facepalm: will test again

Reply:
works